jenkins_pipelines/environments/common/pipeline-mu-aws.groovy

```diff
@@ -37,52 +42,100 @@ def run(params) {
             sh "set +x; source /home/jenkins/.credentials set -x; ./terracumber-cli ${aws_common_params} --gitrepo ${params.sumaform_gitrepo} --gitref ${params.sumaform_ref} --runstep gitsync --sumaform-backend aws"
         }
 
+
         parallel(
-                "create_local_mirror_with_mu": {
-                    stage("Create local mirror with MU") {
-                        writeFile file: "custom_repositories.json", text: params.custom_repositories, encoding: "UTF-8"
-                        mu_repositories = sh(script: "cat ${WORKSPACE}/custom_repositories.json | jq -r ' to_entries[] |  \" \\(.value)\"' | jq -r ' to_entries[] |  \" \\(.value)\"'",
+
+            "upload_latest_image": {
+                if (params.use_latest_ami_image) {
+                    stage('Clean old images') {
+                        // Get all image ami ids
+                        image_amis = sh(script: "${awscli} ec2 describe-images --filters 'Name=name,Values=SUSE-Manager-*-BYOS*' --region ${params.aws_region} | jq -r '.Images[].ImageId'",
+                                returnStdout: true)
+                        // Get all snapshot ids
+                        image_snapshots = sh(script: "${awscli} ec2 describe-images --filters 'Name=name,Values=SUSE-Manager-*-BYOS*' --region ${params.aws_region} | jq -r '.Images[].BlockDeviceMappings[0].Ebs.SnapshotId'",
                                 returnStdout: true)
-                        String[] REPOSITORIES_LIST = mu_repositories.split("\n")
-                        // Create simplify minima file to only synchronize MU
-                        repositories = "storage:\n" +
-                                "  type: file\n" +
-                                "  path: /srv/mirror\n" +
-                                "\n" +
-                                "http:"
-                        REPOSITORIES_LIST.each { item ->
-                            repositories = "${repositories}\n\n" +
-                                    "  - url: ${item}\n" +
-                                    "    archs: [x86_64]"
-                        }
-                        writeFile file: "${local_mirror_dir}/salt/mirror/etc/minima-customize.yaml", text: repositories, encoding: "UTF-8"
 
-                        // Deploy local mirror
-                        sh "set +x; source /home/jenkins/.credentials set -x; export TF_VAR_CUCUMBER_GITREPO=${params.cucumber_gitrepo}; export TF_VAR_CUCUMBER_BRANCH=${params.cucumber_ref}; export TERRAFORM=${params.terraform_bin}; export TERRAFORM_PLUGINS=${params.terraform_bin_plugins}; ./terracumber-cli ${local_mirror_params} --logfile ${resultdirbuild}/sumaform-local.log ${TERRAFORM_INIT} --taint '.*(domain|main_disk).*' --runstep provision --sumaform-backend libvirt"
-                        deployed_local = true
+                        String[] AMI_LIST = image_amis.split("\n")
```

Comment:
Why are you using uppercase here?

Reply:
If you don't use it in more than one line, I would not bother to store it in a variable, just use fluent pattern.